cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/ETLPartitionedFileSetSourceSinkTest.java

```diff
@@ -0,0 +1,176 @@
+    MockAdapterConfigurer adapterConfigurer = new MockAdapterConfigurer();
+    appTemplate.configureAdapter("myAdapter", adapterConfig, adapterConfigurer);
+
+    Map<String, String> mapReduceArgs = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : adapterConfigurer.getArguments().entrySet()) {
+      mapReduceArgs.put(entry.getKey(), entry.getValue());
+    }
+
+    mapReduceArgs.put("config", GSON.toJson(adapterConfig));
+    MapReduceManager mrManager = batchManager.startMapReduce("ETLMapReduce", mapReduceArgs);
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+    batchManager.stopAll();
+
+    verySinkData();
+  }
+
+  private void verySinkData() throws Exception {
+    DataSetManager<PartitionedFileSet> sink = getDataset(sinkFileset);
+    PartitionedFileSet sinkPFS = sink.get();
+    Partition sinkPartition = sinkPFS.getPartition(PartitionKey.builder()
+                                                     .addStringField("league", "nfl")
+                                                     .build());
+
+    Assert.assertNotNull("Partition not found", sinkPartition);
+    try {
+      Location location = sinkPartition.getLocation();
+      for (Location file : location.list()) {
+        if (file.getName().startsWith("part")) {
+          BufferedReader reader = new BufferedReader(new InputStreamReader(location.getInputStream()));
+          Assert.assertEquals("The data in source and sink is not same", reader.readLine(), DUMMY_RECORD);
+        }
+      }
+    } catch (IOException e) {
+      LOG.info("Unable to read path {}", sinkPartition.getRelativePath());
+    }
+  }
+
+  private void writeToSource() throws Exception {
+    DataSetManager<PartitionedFileSet> table1 = getDataset(sourceFileset);
+    PartitionedFileSet inputFileset = table1.get();
+
+    PartitionKey key = PartitionKey.builder()
+      .addStringField("league", "nfl")
+      .addIntField("season", 1985)
+      .build();
+
+    if (inputFileset.getPartition(key) != null) {
+      LOG.info("Partition already exists");
+      throw new RuntimeException("Partition already exists");
+    }
+    PartitionOutput output = inputFileset.getPartitionOutput(key);
+
+    try {
+      Location location = output.getLocation();
+      OutputStream outputStream = location.getOutputStream();
+      try {
+        outputStream.write(DUMMY_RECORD.getBytes(Charsets.UTF_8));
+      } finally {
+        outputStream.close();
+      }
+    } catch (IOException e) {
+      LOG.warn(String.format("Unable to write path '%s'", output.getRelativePath()));
+      throw e;
+    }
+    output.addPartition();
+    table1.flush();
+  }
+
+  private ETLBatchConfig constructETLBatchConfig() {
+    ETLStage source = new ETLStage(PartitionedFileSetSource.class.getSimpleName(),
+                                   ImmutableMap.of("filesetName", sourceFileset, "filter",
+                                                   "{\"league.value\": \"nfl\",\"season.lower\": " +
```

Comment:
Instead of hand writing Json, why not use Gson?

```java
GSON.toJson(ImmutableMap.of("league.value", "nfl", "season.lower", "1980", "season.upper", "1990"))
```